### PR TITLE
7796.master - migration from csv

### PR DESCRIPTION
**Pull request by ivandudardb**
installed the necessary modules
created a .yml file to describe the migration of users to the default table
added password hashing 
created and used a plugin to migrate to a custom database table

### CODE REVIEW

File: web/modules/custom/migrations_csv/src/Plugin/migrate/destination/CustomMigratePlugin.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    $country = $row->getDestinationProperty('country');
+    $interested_in = $row->getDestinationProperty('interested_in');
+    $this->connection->insert('custom_registration_data')
+      ->fields([
+        'uid' => $uid,
+        'city' => $city,
+        'country' => $country,
+        'interested_in' => $interested_in,
+      ])
+      ->execute();
```
Comment (HeorhiiBoliuk) [inline web/modules/custom/migrations_csv/src/Plugin/migrate/destination/CustomMigratePlugin.php:76]: I think that you need to use the primary key here.

**Review by HeorhiiBoliuk — CHANGES_REQUESTED** [review]

**Review by HeorhiiBoliuk — APPROVED** [review]
LGTM

File: web/modules/custom/migrations_csv/src/Plugin/migrate/destination/CustomMigratePlugin.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+   * Constructs a new CustomMigratePlugin object.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration, protected Connection $connection) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $migration);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration = NULL): CustomMigratePlugin|ContainerFactoryPluginInterface|static {
```
Comment (lavrenchuk-dev) [inline web/modules/custom/migrations_csv/src/Plugin/migrate/destination/CustomMigratePlugin.php]: ?

File: web/modules/custom/migrations_csv/src/Plugin/migrate/destination/CustomMigratePlugin.php (hunk 1 above)
Comment (lavrenchuk-dev) [reply web/modules/custom/migrations_csv/src/Plugin/migrate/destination/CustomMigratePlugin.php:89]: Don't you have a service to work with your custom table?

**Review by lavrenchuk-dev — CHANGES_REQUESTED** [review]

**Review by lavrenchuk-dev — APPROVED** [review]

**Pull request merged**